.gitignore

```diff
@@ -0,0 +1,4 @@
+#inteliJ file type
+*.iml
+#inteliJ directory
+.idea/
```

Comment:
We may want to add some of these for java as well:
Reference:https://github.com/github/gitignore/blob/master/Java.gitignore
```
# Compiled class file
*.class

# Log file
*.log

# BlueJ files
*.ctxt

# Mobile Tools for Java (J2ME)
.mtj.tmp/

# Package Files #
*.jar
*.war
*.ear
*.zip
*.tar.gz
*.rar

# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
hs_err_pid*
```